### ISSUE TITLE
Config options for directory sharing

### ISSUE THREAD

**Issue by anjannath** [code-ready/crc#3310]
- `enable-shared-dirs` 
this was added to toggle the automounting of shared directories during start.

for windows CIFS/SMB file sharing we need the user's login credentials to mount the shared directory, which creates the need for another config option
- `smb-share-password`

when we add support for `sshfs` we need another config option to allow users to use the `sshfs` based file sharing instead of the platform default method

- `use-rsshfs-dir-share`

**Comment by gbraad** [comment]
you can not use a single option for a share. each share can have it's own set of variables.

`host path`, `mount path`, `type`, `options`/`password`

**Comment by anjannath** [comment]
> you can not use a single option for a share. each share can have it's own set of variables.
> 
> `host path`, `mount path`, `type`, `options`/`password`
> 

for CRC these options doesn't need to be configurable as we want to provide the user the same behaviour as `podman` does, i.e mount the user's home directory in the vm at `/home/$username` or in case of windows at `/mnt/c/users/$username`

so, 
- the `host_path` in our case is always going to be the user's home folder
- mount path defaults to `/mnt/c/user/$username`
- options passed with `-o` for mounting should be internal and user doesn't need to think or configure these

We just want the `password` to be configurable and only for CIFS so the `smb-share-password` config will only exist on windows

**Comment by gbraad** [comment]
>  for mounting should be internal and user doesn't need to think or configure these

if you aren't considering using hostfolders other than from 127.0.0.1

**Comment by anjannath** [comment]
right, only considering the local use case for now, for a remote like scenario `sshfs` would be better

**Comment by gbraad** [comment]
you however have to take into account to use similar, if not same, configuration/command line options to achieve this. you can not just easily change them once released. and you can have another set of commands to deal with another type, this is confusing.

**Comment by anjannath** [comment]
If we add a config `smb-share-password` now for CIFS based share, we can add a similar `sshfs-share-password` later for the `sshfs` base share, or are you suggesting to use a generic config, e.g `share-password`

since we are not adding a config for `Source Path` and `Target Path` now for CIFS, we can defer the decision regarding these config's names a bit.
